src/utils/audio-cache.ts

```
@@ -0,0 +1,289 @@
+      headers: {
+        'Content-Type': 'audio/mpeg',
+        'Content-Length': fileSize.toString(),
+      },
+    });
+
+    // Store in cache storage
+    const cache = await caches.open(CACHE_NAME);
+    await cache.put(url, blobResponse);
+
+    // Store metadata in IndexedDB
+    const db = await getDB();
```

ESLint

[error] The variable `db` should be named `database`. A more descriptive name will do too.

(unicorn/prevent-abbreviations)